test/spatial_reaction_systems/lattice_reaction_systems_jumps.jl

```diff
@@ -0,0 +1,309 @@
+                    pE_4 = make_u0_matrix(pE_3, edges(lrs.lattice),
+                                          ModelingToolkit.getname.(diffusion_parameters(lrs)))
+                    for pE in [pE_1, pE_2, pE_3, pE_4]
+                        dprob = DiscreteProblem(lrs, u0, (0.0, 100.0), (pV, pE))
+                        jprob = JumpProblem(lrs, dprob, NSM())
+                        solve(jprob, SSAStepper())
+                    end
+                end
+            end
+        end
+    end
+end
+
+### Input Handling Tests ###
+
+# Tests that the correct hopping rates and initial conditions are generated.
+# In this base case, hoppping rates should be on the form D_{s,i,j}.
+let
+    # Prepares the system.
+    lrs = LatticeReactionSystem(SIR_system, SIR_srs_2, small_2d_grid)
+
+    # Prepares various u0 input types.
+    u0_1 = [:I => 2.0, :S => 1.0, :R => 3.0]
+    u0_2 = [:I => fill(2., nv(small_2d_grid)), :S => 1.0, :R => 3.0]
+    u0_3 = [1.0, 2.0, 3.0]
+    u0_4 = [1.0, fill(2., nv(small_2d_grid)), 3.0]
+    u0_5 = permutedims(hcat(fill(1., nv(small_2d_grid)), fill(2., nv(small_2d_grid)), fill(3., nv(small_2d_grid))))
+
+    # Prepare various (compartment) parameter input types.
+    pC_1 = [:β => 0.2, :α => 0.1]
+    pC_2 = [:β => fill(0.2, nv(small_2d_grid)), :α => 1.0]
+    pC_3 = [0.1, 0.2]
+    pC_4 = [0.1, fill(0.2, nv(small_2d_grid))]
+    pC_5 = permutedims(hcat(fill(0.1, nv(small_2d_grid)), fill(0.2, nv(small_2d_grid))))
+
+    # Prepare various (diffusion) parameter input types.
+    pD_1 = [:dI => 0.02, :dS => 0.01, :dR => 0.03]
+    pD_2 = [:dI => 0.02, :dS => fill(0.01, ne(small_2d_grid)), :dR => 0.03]
+    pD_3 = [0.01, 0.02, 0.03]
+    pD_4 = [fill(0.01, ne(small_2d_grid)), 0.02, 0.03]
+    pD_5 = permutedims(hcat(fill(0.01, ne(small_2d_grid)), fill(0.02, ne(small_2d_grid)), fill(0.03, ne(small_2d_grid))))
+
+    # Checks hopping rates and u0 are correct.
+    true_u0 = [fill(1.0, 1, 25); fill(2.0, 1, 25); fill(3.0, 1, 25)]
+    true_hopping_rates = cumsum.([fill(dval, length(v)) for dval in [0.01,0.02,0.03], v in small_2d_grid.fadjlist])
+    for u0 in [u0_1, u0_2, u0_3, u0_4, u0_5]
+        # Provides parameters as a tupple.
+        for pC in [pC_1, pC_3], pD in [pD_1, pD_2, pD_3, pD_4, pD_5]
+            dprob = DiscreteProblem(lrs, u0, (0.0, 100.0), (pC,pD))
+            jprob = JumpProblem(lrs, dprob, NSM())
+            @test jprob.prob.u0 == true_u0
+            @test jprob.discrete_jump_aggregation.hop_rates.hop_const_cumulative_sums == true_hopping_rates
+        end
+        # Provides parameters as a combined vector.
+        for pC in [pC_1], pD in [pD_1, pD_2]
+            dprob = DiscreteProblem(lrs, u0, (0.0, 100.0), [pD; pC])
+            jprob = JumpProblem(lrs, dprob, NSM())
+            @test jprob.prob.u0 == true_u0
+            @test jprob.discrete_jump_aggregation.hop_rates.hop_const_cumulative_sums == true_hopping_rates
```

Comment:
Let's also test that the massaction jumps are correct, along with the hopping rates.

Reply:
I was meaning to, but ran into some weird problem, will see if I can recreate what happened.

Reply:
Added tests for this as well